torch/lib/THC/generic/THCTensorMathPointwise.cu

```diff
@@ -61,6 +60,13 @@ IMPLEMENT_CUDA_TENSOR_BASIC_FUNC( cinv, THCNumerics<real>::cinv,  Real)
 
 #endif
 
+#if defined(THC_REAL_IS_FLOAT) || defined(THC_REAL_IS_DOUBLE) || defined(THC_REAL_IS_HALF) || \
```

Comment:
I don't really have a good reason why things are this way; it seems like e.g. abs is defined on char cuda tensors, but not char tensors; rather than make this issue larger (this is to fix a breaking change for some people), I filed https://github.com/pytorch/pytorch/issues/2359 where we can figure it out separately.